dataclients/kubernetes/crd.go

```diff
@@ -0,0 +1,171 @@
+package kubernetes
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/zalando/skipper/eskip"
+)
+
+type routeGroupList struct {
+	Items []*routeGroupItem `json:"items"`
+}
+
+type routeGroupItem struct {
+	Metadata *metadata       `json:"metadata"`
+	Spec     *routeGroupSpec `json:"spec"`
+}
+
+type routeGroupSpec struct {
+	Hosts    []string          `json:"hosts,omitempty"`
+	Backends []*skipperBackend `json:"backends"`
+	Paths    []*pathSpec       `json:"paths"`
+}
+
+type pathSpec struct {
+	Path       string   `json:"path"`
+	Method     string   `json:"method,omitempty"`
+	Filters    []string `json:"filters,omitempty"`
+	Predicates []string `json:"predicates,omitempty"`
+	Backend    string   `json:"backend,omitempty"`
+}
+
+type algorithm int
+
+const (
+	none algorithm = iota
+	roundrobin
+	random
+	consistentHash
+)
+
+type skipperBackendParser struct {
+	Name        string   `json:"name"`
+	Type        string   `json:"type"`
+	ServiceName string   `json:"serviceName"`
+	ServicePort int      `json:"servicePort"`
+	Default     bool     `json:"default"`
+	Algorithm   string   `json:"algorithm"`
+	Endpoints   []string `json:"endpoints"`
+}
+
+// can be:
+// - *backend defined in definitions.go
+// - SpecialBackend string   // <shunt>, ..
+type skipperBackend struct {
+	Name        string
+	Type        eskip.BackendType
+	Default     bool
+	ServiceName string
+	ServicePort int
+	Algorithm   algorithm
+	Endpoints   []string
+}
+
+// adding Kubernetes specific backend types here. To be discussed.
+// The main reason is to differentiate between service and external, in a way
+// where we can also use the current global option to decide whether the service
+// should then be converted to LB. Or shall we expect the route group already
+// contain the pod endpoints, and ignore the global option for skipper?
+const (
+	serviceBackend = eskip.LBBackend + iota
+	externalURL
+)
+
+// Shall we support dynamic backend? Does it make sense in this
+// scenario?
+func backendTypeFromString(s string) (eskip.BackendType, error) {
+	switch s {
+	case "", "service":
+		return serviceBackend, nil
+	case "external":
+		return externalURL, nil
```

Comment:
why not eskip.NetworkBackend?

Reply:
It doesn't have the same meaning. When we generate the actual routes, and a service uses ClusterIP, then both the external and cluster IP would be network backends in eskip terms. I wanted to avoid mixing them here. It's just a package internal temporary phase currently. Would it still be better to just go and use eskip.NetworkBackend?

Reply:
Good question, I think at service parse time it makes sense to differentiate between service types, on intermediate object to create routes, I would prefer the skipper specific names.